MediaBrowser.Providers/Plugins/Tmdb/TV/TmdbEpisodeProvider.cs

```diff
@@ -102,9 +104,48 @@ public async Task<MetadataResult<Episode>> GetMetadata(EpisodeInfo info, Cancell
                 return metadataResult;
             }
 
-            var episodeResult = await _tmdbClientManager
-                .GetEpisodeAsync(seriesTmdbId, seasonNumber.Value, episodeNumber.Value, info.SeriesDisplayOrder, info.MetadataLanguage, TmdbUtils.GetImageLanguagesParam(info.MetadataLanguage), cancellationToken)
-                .ConfigureAwait(false);
+            var episodeResult = new TvEpisode();
+            if (!info.IndexNumberEnd.HasValue)
+            {
+                episodeResult = await _tmdbClientManager
+                    .GetEpisodeAsync(seriesTmdbId, seasonNumber.Value, episodeNumber.Value, info.SeriesDisplayOrder, info.MetadataLanguage, TmdbUtils.GetImageLanguagesParam(info.MetadataLanguage), cancellationToken)
+                    .ConfigureAwait(false);
+            }
+            else
+            {
+                var startindex = episodeNumber;
+                var endindex = info.IndexNumberEnd;
+                List<TvEpisode> result = new List<TvEpisode>();
+                for (int? episode = startindex; episode <= endindex; episode++)
+                {
+                    var episodeInfo = await _tmdbClientManager.GetEpisodeAsync(seriesTmdbId, seasonNumber.Value, episode.Value, info.SeriesDisplayOrder, info.MetadataLanguage, TmdbUtils.GetImageLanguagesParam(info.MetadataLanguage), cancellationToken).ConfigureAwait(false);
+                    if (episodeInfo is not null)
+                    {
+                        result.Add(episodeInfo);
+                    }
+                }
+
+                if (result.Count > 0)
+                {
+                    episodeResult = result[0];
+                }
+                else
+                {
+                    return metadataResult;
+                }
+
```

Comment:
Should we add a fast return if only 1 result was found?

Reply:
I can shift the following code 

                var name = new StringBuilder(episodeResult.Name);
                var overview = new StringBuilder(episodeResult.Overview);

                for (int i = 1; i < result.Count; i++)
                {
                    name.Append(" / " + result[i].Name);
                    overview.Append(" / " + result[i].Overview);
                }

                episodeResult.Name = name.ToString();
                episodeResult.Overview = overview.ToString();

to be under

                if (result is not null)

and do a check if there is only one result. Then, can exit the if statment ealier without going through the for loop?